### PR TITLE
test(cli): Add monorepo tests

### LINKED ISSUE #36: Test monorepo
(free for assignment to any raid member)

From coverage report of `meta.ts`, we need to test a project with "monorepo" structure (not sure what it means exactly, if not obvious by analyzing code, we will need to ask Stephan):

Screenshot 2021-01-28 at 13 40 23

### PR DESCRIPTION
**Pull request by datner**
closes: #36

testing the ts paths one was pretty easy, but for the monorepo support I had to innovate.
add optional directories argument to create directories
add optional baseDir to change the base dir to something else.

names used in tests either corresponds to names used in implementation code, or generic placeholder / foobarbaz names

### CODE REVIEW

File: src/__tests__/cli.ts [hunk 1]
```diff
@@ -153,12 +161,52 @@ export default promise
...
   scenarios.forEach((scenario) => {
     test(scenario.description, async () => {
-      const testProjectDir = await createProject(scenario.files);
+      const testProjectDir = await createProject(
+        scenario.files,
+        scenario.directories,
+        scenario.baseDir,
+      );
 
+      console.log(testProjectDir);
```
Comment (mpeyper) [inline src/__tests__/cli.ts]: left one in :wink:

Comment (datner) [reply src/__tests__/cli.ts]: oops

Comment (datner) [reply src/__tests__/cli.ts]: done

File: src/__tests__/cli.ts [hunk 2]
```diff
@@ -60,20 +60,28 @@ async function exec(
...
 
   const testSpaceDir = path.join('.test-space', randomId.toString());
 
   await mkdir(testSpaceDir, { recursive: true });
 
+  await Promise.all(
+    directories.map((dir) =>
+      mkdir(path.join(testSpaceDir, dir), { recursive: true }),
+    ),
+  );
```
Comment (mpeyper) [inline src/__tests__/cli.ts]: I don't mind this, but I worry that the scenario setup could get fiddly as maintenance progresses in the future.  Having it pick up the directories from the file names using [`path.dirname`](https://nodejs.org/api/path.html#path_path_dirname_path) could make it a little bit more automatic and less to worry about without changing the implementation here too much. Something like:

```ts
  await Promise.all(
    files
      .map((file) => path.dirname(file.name))
      .map((dir) => mkdir(path.join(testSpaceDir, dir), { recursive: true })),
  );
  ```

The you can do away with the `directories` here and in the scenario config.

**Review by Aprillion — APPROVED** [review]

File: src/__tests__/cli.ts (hunk 2 above)
Comment (datner) [reply src/__tests__/cli.ts]: oh, didn't know about `path.dirname`
changing

Comment (datner) [reply src/__tests__/cli.ts:76]: with this I can squash it with the other Promise.all

Comment (datner) [reply src/__tests__/cli.ts]: done

**Review by Aprillion — APPROVED** [review]

**Pull request merged**